src/Admin/Input/Integer.php

```diff
@@ -20,6 +20,15 @@ public function __construct() {
 		// the text field block to work around it.
 		parent::__construct( 'integer', 'woocommerce/product-text-field' );
 
-		$this->set_block_attribute( 'type', [ 'value' => 'number' ] );
+		$this->set_block_attribute(
+			'pattern',
+			[
+				'value'   => '\d+',
+				// This error message is used in core so to avoid needing to translate
+				// the string a second time WooCommerce is supplied as the text domain.
+				// phpcs:ignore WordPress.WP.I18n.TextDomainMismatch
+				'message' => __( 'Please enter a valid value.', 'woocommerce' ),
```

Comment:
I couldn't find the same text in WooCommerce.

Is it supposed to be "Invalid value for the field."? Or perhaps leave it unspecified and let it fall back to the default error message in WooCommerce.

Reply:
I couldn't find that text either, so I removed the message and switched to the fallback.: [e40b122](https://github.com/woocommerce/google-listings-and-ads/pull/2511/commits/e40b1225afb555a74e8a959de3cd008c578824fc)